File: words-game/words.py
import random
import os

with open('usa.txt') as file:
    lst = [line.strip() for line in file]


char_len = int(input("Enter length of word (3-12): "))
while char_len not in range(3, 13):
    char_len = int(input("Enter length of word (3-12): "))
    
word = lst[random.randint(0,len(lst))]
while len(word) != char_len:
    word = lst[random.randint(0,len(lst))]


pattern = list('-' * char_len)

def indexes(s, c):
    ins = []
    for i in range(len(s)):
        if s[i] == c:
            ins.append(i)
    return ins

def insert_ins(p, ins, c):
    for i in ins:
        p[i] = c
    return p

#GAME_LOOP
tries_count = 0
while pattern != list(word):
    os.system('cls')
    print(''.join(pattern))
    char = input("Enter char to guess: ")
    while len(char) != 1:
        char = input("Pleaser enter 1 valid character: ")
            
    ins = indexes(word, char)
    pattern = insert_ins(pattern, ins, char)
    
    tries_count += 1
    
os.system('cls')
print(f"Congrats, you did this at {tries_count}'s try")
print(word)

File: words-game/words.rs
use rand::seq::SliceRandom;
use std::fs;
use std::io::{self, Write};
use std::process::Command;

fn main() {
    let contents = fs::read_to_string("resources/usa.txt").expect("Cannot read file");
    let lst: Vec<&str> = contents.lines().map(str::trim).collect();

    let mut char_len = input("Enter length of word (3-12): ")
        .parse::<usize>().unwrap_or(0);

    while !(3..=12).contains(&char_len) {
        char_len = input("Enter length of word (3-12): ")
            .parse::<usize>().unwrap_or(0);
    }

    let mut rng = rand::thread_rng();
    let mut word = lst.choose(&mut rng).unwrap_or(&"").to_string();

    while word.len() != char_len {
        word = lst.choose(&mut rng).unwrap_or(&"").to_string();
    }

    let mut pattern: Vec<char> = "-".repeat(char_len).chars().collect();

    let mut tries_count = 0;
    while pattern.iter().collect::<String>() != word {
        clear_screen();
        println!("{}", pattern.iter().collect::<String>());
        let mut char = input("Enter char to guess: ");
        while char.len() != 1 {
            char = input("Pleaser enter 1 valid character: ");
        }

        let c = char.chars().next().unwrap();
        let ins = indexes(&word, c);
        pattern = insert_ins(pattern, &ins, c);

        tries_count += 1;
    }

    clear_screen();
    println!("Congrats, you did this at {}'s try", tries_count);
    println!("{}", word);
}

fn indexes(s: &str, c: char) -> Vec<usize> {
    s.chars()
        .enumerate()
        .filter_map(|(i, ch)| if ch == c { Some(i) } else { None })
        .collect()
}

fn insert_ins(mut p: Vec<char>, ins: &[usize], c: char) -> Vec<char> {
    for &i in ins {
        p[i] = c;
    }
    p
}

fn input(prompt: &str) -> String {
    print!("{}", prompt);
    io::stdout().flush().unwrap();
    let mut buf = String::new();
    io::stdin().read_line(&mut buf).unwrap();
    buf.trim().to_string()
}

fn clear_screen() {
    Command::new("clear").status().unwrap();
}
